### PR TITLE
Reconstruct Speculative Decoding example directory

### PR DESCRIPTION
**Pull request by jenniew**
Reorganize the Speculative Decoding example directory. 
Make example/CPU/Speculative-Decoding with Self-Speculation and Eagle folders
Put ipex-llm self speculative example to Self-Speculation directory.
Put Eagle examples to Eagle folder

### CODE REVIEW

File: python/llm/example/GPU/Speculative-Decoding/README.md [hunk 1]
```diff
@@ -1,26 +1,6 @@
...
-## Best Known Configuration on Linux
-
-For optimal performance on Intel Data Center GPU Max Series, it is recommended to set several environment variables.
-```bash
-export LD_PRELOAD=${LD_PRELOAD}:${CONDA_PREFIX}/lib/libtcmalloc.so
-export SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1
-export ENABLE_SDP_FUSION=1
-```
+- [Self-Speculation](Self-Speculation): running BF16 inference for Huggingface Transformer model with ***self-speculative decoding*** with IPEX-LLM on Intel GPUs
+- [Eagel](Eagle): running speculative sampling using EAGLE (Extrapolation Algorithm for Greater Language-model Efficiency) with IPEX-LLM on Intel GPUs
```
Comment (jason-dai) [inline python/llm/example/GPU/Speculative-Decoding/README.md:6]: make EAGLE bold

Comment (jenniew) [reply python/llm/example/GPU/Speculative-Decoding/README.md]: fixed

File: python/llm/example/GPU/Speculative-Decoding/Self-Speculation/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Self-Speculative Decoding for Large Language Model FP16 Inference using IPEX-LLM on Intel GPUs
+You can use IPEX-LLM to run FP16 inference for any Huggingface Transformer model with ***self-speculative decoding*** on Intel GPUs. This directory contains example scripts to help you quickly get started to run some popular open-source models using self-speculative decoding. Each model has its own dedicated folder, where you can find detailed instructions on how to install and run it.
+
+## Verified Hardware Platforms
+
+- Intel Data Center GPU Max Series
+
+## Recommended Requirements
+To apply Intel GPU acceleration, there’re several steps for tools installation and environment preparation. See the [GPU installation guide](https://ipex-llm.readthedocs.io/en/latest/doc/LLM/Overview/install_gpu.html) for mode details.
```
Comment (rnwang04) [inline python/llm/example/GPU/Speculative-Decoding/Self-Speculation/README.md:9]: for **more** details.

**Review by rnwang04 — APPROVED** [review]
others LGTM.

File: python/llm/example/GPU/Speculative-Decoding/Self-Speculation/README.md (hunk 2 above)
Comment (jenniew) [reply python/llm/example/GPU/Speculative-Decoding/Self-Speculation/README.md]: fixed

**Pull request merged**